test/integration/change-streams/change_stream.test.ts

```diff
@@ -2551,6 +2635,40 @@ describe('ChangeStream resumability', function () {
           }
         );
       });
+
+      context('when the error occurs on the aggregate command', function () {
+        it(
+          'does not resume',
+          { requires: { topology: '!single', mongodb: '>=4.2' } },
+          async function () {
+            changeStream = collection.watch([]);
+
+            const docs = [{ city: 'New York City' }, { city: 'Seattle' }, { city: 'Boston' }];
+            await collection.insertMany(docs);
+
+            const resumableErrorCode = 7;
+            await client.db('admin').command({
+              configureFailPoint: 'failCommand',
+              mode: { times: 2 }, // Account for retry in executeOperation which is separate from change stream's resume
+              data: {
+                failCommands: ['aggregate'],
+                errorCode: resumableErrorCode
+              }
+            } as FailPoint);
+
+            try {
+              // eslint-disable-next-line @typescript-eslint/no-unused-vars
+              for await (const change of changeStream) {
+                expect.fail('Change stream produced events on an unresumable error');
+              }
```

Comment:
```suggestion
              for await (const change of changeStream);
              expect.fail('Change stream produced events on an unresumable error');              
```

Right? just to catch a case where no iteration and no error happens?

Reply:
The intention here was to throw an error when a change stream is _successfully_ iterable after an unresumable error, but throwing on the case where it doesn't iterate is something that was missing here.